### PR TITLE
Moving Fidget Settings to Edit Panel

### PR DESCRIPTION
**Pull request by j-paterson**

### CODE REVIEW

File: src/common/components/organisms/EditorPanel.tsx [hunk 1]
```diff
@@ -1,33 +1,86 @@
...
 
 export interface EditorPanelProps {
-  editMode: boolean;
   setEditMode: (editMode: boolean) => void;
   theme?: ThemeSettings;
   saveTheme: (newTheme: ThemeSettings) => void;
-  disabled?: boolean;
+  unselect: () => void;
+  selectedFidgetID: string | null;
+  currentFidgetSettings: React.JSX.Element;
```
Comment (hiporox) [inline src/common/components/organisms/EditorPanel.tsx]: This should be `React.ReactNode` to allow it to properly be a `null` or `string` value in the event that no `currentFidget` is set

File: src/common/components/organisms/EditorPanel.tsx [hunk 2]
```diff
@@ -1,33 +1,86 @@
...
+          <div className="h-full px-4 py-4 overflow-y-auto border border-blue-100 rounded-xl relative bg-card">
+            <div className="flex-col">
+              {selectedFidgetID ? (
+                <>
+                  <div className="flex">
+                    <button onClick={unselect} className="my-auto">
+                      <BackArrowIcon />
+                    </button>
+                    <h1 className="capitalize text-lg pl-4">
+                      Edit {selectedFidgetID}
```
Comment (hiporox) [inline src/common/components/organisms/EditorPanel.tsx]: This is fine for now, but Fidget ID will not be a nice human readable thing down the line

File: src/common/components/organisms/EditorPanel.tsx [hunk 3]
```diff
@@ -1,33 +1,86 @@
...
+        </div>
+      </aside>
+      <div className="w-4/12">
+        <FidgetTray />
+      </div>
+    </div>
+  );
+};
+
+const BackArrowIcon = () => {
```
Comment (hiporox) [inline src/common/components/organisms/EditorPanel.tsx]: These Icons really need to be moved to their own files

File: src/common/components/organisms/FidgetTray.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
+import { FidgetConfig, FidgetSettings } from "@/common/fidgets";
+import React from "react";
+
+const PlusIcon = () => {
```
Comment (hiporox) [inline src/common/components/organisms/FidgetTray.tsx:4]: Seriously, please make all of these Icons go in their own icons folder

File: src/common/fidgets/FidgetWrapper.tsx [hunk 5]
```diff
@@ -73,33 +88,36 @@ export function FidgetWrapper({
       toast.error("Failed to save fidget settings", { duration: 1000 });
     }
     setSaving(false);
+    setEditing(false);
+    setSelectedFidgetID("");
```
Comment (hiporox) [inline src/common/fidgets/FidgetWrapper.tsx:91]: Call `unselect` here

File: src/common/fidgets/FidgetWrapper.tsx [hunk 6]
```diff
@@ -13,12 +13,16 @@ import {
...
 import FidgetSettingsPopover from "@/common/fidgets/FidgetSettingsPopover";
 import { reduce } from "lodash";
+import FidgetSettingsEditor from "../components/organisms/FidgetSettingsEditor";
 
 export type FidgetWrapperProps = {
   fidget: React.FC<FidgetArgs>;
   config: FidgetDetails;
   context?: FidgetRenderContext;
   saveConfig: (conf: FidgetConfig) => Promise<void>;
+  setcurrentFidgetSettings: (currentFidgetSettings: React.JSX.Element) => void;
```
Comment (hiporox) [inline src/common/fidgets/FidgetWrapper.tsx]: These three should be rolled up into `onEnterEditMode` + `onExitEditMode` functions that the parent can control how they works. The wrapper doesn't need to have a concept of the selection process, only that it is supposed to call `onEnterEditMode` with a bunch of information (fidget ID + Editor Component) each time if enters edit mode and call `onExitEditMode` each time it leaves

**Comment by hiporox** [comment]
You could actually portal the FidgetWrapper edit mode into the sidebar. That is probably the most correct behavior: https://react.dev/reference/react-dom/createPortal

Something along the lines of an optional arg passed to the FidgetWrapper that is the dom node to portal to, and if it is null, you just render the edit mode in place of the Fidget, but if it is present, you can pass the edit mode in a portal to the provided dom node

This set up keeps FidgetWrapper as layout independent as possible

File: src/common/fidgets/FidgetWrapper.tsx (hunk 5 above)
Comment (hiporox) [reply src/common/fidgets/FidgetWrapper.tsx:91]: In fact, all of this should be rolled  up into a `onExitEditMode` function that is provided

File: src/common/components/organisms/EditorPanel.tsx (hunk 2 above)
Comment (j-paterson) [reply src/common/components/organisms/EditorPanel.tsx]: Yeah we do need to replace this with a 'Fidget Name' field.

File: src/common/components/organisms/FidgetTray.tsx (hunk 4 above)
Comment (j-paterson) [reply src/common/components/organisms/FidgetTray.tsx:4]: On it.

File: src/common/components/organisms/Sidebar.tsx [hunk 7]
```diff
@@ -1,5 +1,5 @@
 import { ThemeSettings } from "@/common/lib/theme";
-import React from "react";
+import React, { ReactNode } from "react";
```
Comment (hiporox) [inline src/common/components/organisms/Sidebar.tsx:2]: Unnecessary extra import

File: src/common/components/organisms/Navigation.tsx [hunk 8]
```diff
@@ -1,31 +1,93 @@
...
+            </div>
+          </div>
+        </div>
+      </div>
+    </aside>
   );
 };
 
-const FeedIcon = () => {
+const HomeIcon = () => {
```
Comment (hiporox) [inline src/common/components/organisms/Navigation.tsx:90]: Icons still in this file

**Review by hiporox — APPROVED** [review]
On the whole good.

I would still like to see Icons in their own folder, portals for the `FidgetWrapper`, and rolling `setcurrentFidgetSettings, setSelectedFidgetID, selectedFidgetID` into a a handler method on `FidgetWrapper` instead of it knowing  about the selection process. But all of these are things we can do in a future PR and can be added to the backlog for now

File: src/common/fidgets/FidgetWrapper.tsx (hunk 6 above)
Comment (hiporox) [inline src/common/fidgets/FidgetWrapper.tsx]: This should also be a ReactNode not a JSX.Element as well

**Pull request merged**